index.js

```diff
@@ -20,6 +20,9 @@ export default function createCachedSelector(...funcs) {
             delete cache[earliest];
           }
         }
+
+        // Move the latest cache key to the top.
+        cacheOrdering = [...cacheOrdering.filter(key => key !== cacheKey), cacheKey];
```

Comment:
put this in an `else` directly above so that you're not doing this when you just used `push`.